src/core/server/plugins/types.ts

```diff
@@ -93,7 +93,7 @@ export type PluginOpaqueId = symbol;
  */
 export interface PluginManifest {
   /**
-   * Identifier of the plugin.
+   * Identifier of the plugin. Must be a string in snake_case.
```

Comment:
looks like a convention for legacy plugins

Reply:
The downside is then the access path looks weird.  `npStart.plugins.uiActions.getTriggerCompatibleActions` vs `npStart.plugins.ui_actions.getTriggerCompatibleActions`

Reply:
created a separate issue https://github.com/elastic/kibana/issues/52190 to dicsuss